docs/blueprints/storage-node-satellite-selection.md

```diff
@@ -0,0 +1,295 @@
+  subdomain of that host.
+* When the `untrusted` entry is a full Satellite URL, it matches any URL that
+  is equal.
+
+After the `trusted` list has been pruned, the remaining URLs are aggregated
+according to the following rules:
+
+* A Satellite URL is considered _authoritative_ if it matches either of the
+  following criteria:
+    * Configured via a Trusted Satellite URL entry
+    * Configured via a `file://` URL
+    * Configured via an `https://` or `http://` Trusted Satellite List URL AND matches the domain or is a subdomain of the domain name in the Trusted Satellite List URL.
+* Satellite URLs are equivalent if the address portions are equal.
+* When aggregating equivalent Satellite URLs (i.e. address matches) with
+  differing IDs, the _authoritative_ Satellite URL wins. If neither or both are
+  _authoritative_, the one aggregated first wins.
+
+
+#### Example
+
+Consider the following Trusted Satellite List URLs and their contents. For
+brevity sake, the full ID of each URL is being shortened (real configurations
+**MUST** specify the full ID).
+
+* `file:///path/to/some/trusted-satellites.txt`
+
+```
+1@bar.test:7777
+```
+
+* `https://foo.test/trusted-satellites`
+
+```
+2@f.foo.test:7777
+2@buz.test:7777
+2@qiz.test:7777
+5@ohno.test:7777
+```
+
+* `https://bar.test/trusted-satellites`
+
+```
+3@f.foo.test:7777
+3@bar.test:7777
+3@buz.test:7777
+3@quz.test:7777
+```
+
+* `https://baz.test/trusted-satellites`
+
+```
+4@qiz.test:7777
+4@subdomain.quz.test:7777
+```
+
+Now consider the following configuration:
+
+```
+- !quz.test
+- file:///path/to/some/trusted-satellites.txt
+- https://foo.test/trusted-satellites
+- https://bar.test/trusted-satellites
+- https://baz.test/trusted-satellites
+- 0@f.foo.test:7777
+- !2@qiz.test:7777
+- !5
+```
+
+After expanding each entry, we have the following unaggregated `trusted` list:
+
+```
+1@bar.test:7777                   (authoritative due to file:// URL)
+2@f.foo.test:7777                 (authoritative due to foo.test domain)
+2@buz.test:7777
+2@qiz.test:7777
+5@ohno.test:7777
+3@f.foo.test:7777
+3@bar.test:7777                   (authoritative due to bar.test domain)
+3@buz.test:7777
+3@quz.test:7777
+4@qiz.test:7777
+4@subdomain.quz.test:7777
+0@f.foo.test:7777
+```
+
+And the following `untrusted` list:
+
+```
+quz.test
+2@qiz.test:7777
+5
+```
+
+The `trusted` list is pruned with the `untrusted` list, leaving the following `trusted` list:
+
+```
+1@bar.test:7777                   (authoritative due to file:// URL)
+2@f.foo.test:7777                 (authoritative due to foo.test domain)
+2@buz.test:7777
+3@f.foo.test:7777
+3@bar.test:7777                   (authoritative due to bar.test domain)
+3@buz.test:7777
+0@f.foo.test:7777
+4@qiz.test:7777
+```
+
+We aggregate from top to bottom (i.e. in the order they were specified/fetched)
+and are left with the following:
+
+```
+1@bar.test:7777
+2@f.foo.test:7777
+2@buz.test:7777
```

Comment:
And now it's gone.

Reply:
It's supposed to be gone... the list pulled from `foo.test` was authoritative for `f.foo.test` and it came first.